### PR TITLE
#107/feat/node mocks http

### PR DESCRIPTION
**Pull request by mmoila**

### CODE REVIEW

File: src/app/api/course/route.test.ts [hunk 1]
```diff
@@ -34,15 +36,28 @@ describe('API', () => {
 
   describe('POST', () => {
     it('adds new course in to the database', async () => {
-      const response = await POST(newCourse as any);
+      const { req } = createMocks<NextRequest>({
+        method: 'POST',
+        body: newCourse,
+        json: () => newCourse,
```
Comment (DeeCaaD) [inline src/app/api/course/route.test.ts:42]: Body & json both required?

**Review by DeeCaaD — APPROVED** [review]
Looks good!

File: src/app/api/course/route.test.ts (hunk 1 above)
Comment (mmoila) [reply src/app/api/course/route.test.ts:42]: Noup. Just the json function used. Body left there for some unknown reason 🤷‍♂️

File: src/app/api/course/route.test.ts [hunk 2]
```diff
@@ -34,15 +36,28 @@ describe('API', () => {
 
   describe('POST', () => {
     it('adds new course in to the database', async () => {
-      const response = await POST(newCourse as any);
+      const { req } = createMocks<NextRequest>({
+        method: 'POST',
+        body: newCourse,
```
Comment (DeeCaaD) [inline src/app/api/course/route.test.ts:41]:
```suggestion
```

File: src/app/api/course/route.test.ts [hunk 3]
```diff
@@ -34,15 +36,28 @@ describe('API', () => {
...
       expect(data.message).toBe('Course succesfully created!');
       expect(data.messageType).toBe(MessageType.SUCCESS);
       expect(response.status).toBe(201);
     });
 
     it('fails with incorrect inputs', async () => {
-      const response = await POST(failedCourse as any);
+      const { req } = createMocks<NextRequest>({
+        method: 'POST',
+        body: failedCourse,
```
Comment (DeeCaaD) [inline src/app/api/course/route.test.ts:56]:
```suggestion
```

**Review by DeeCaaD — APPROVED** [review]

**Pull request merged**